java/ql/src/Telemetry/ExternalAPI.qll

```diff
@@ -0,0 +1,52 @@
+/** Provides classes and predicates related to handling APIs from external libraries. */
+
+private import java
+private import APIUsage
+private import semmle.code.java.dataflow.ExternalFlow
+
+/**
+ * An external API from either the Java Standard Library or a 3rd party library.
+ */
+class ExternalAPI extends Callable {
+  ExternalAPI() { not this.fromSource() }
+
+  /** Holds true if this API is part of a common testing library or framework */
+  predicate isTestLibrary() { getDeclaringType() instanceof TestLibrary }
+
+  /**
+   * Gets information about the external API in the form expected by the CSV modeling framework.
+   */
+  string asCSV(ExternalAPI api) {
```

Comment:
We've discussed abbreviations in concatenated identifiers (camlCase, PascalCase) before, and I believe we've had reasonable consensus that we'll want to camlCase them as well (at least for 3+ letter abbrev.) even though we aren't being completely consistent at the moment.
```suggestion
  string asCsv(ExternalApi api) {
```